reference/test/matrix/csr_kernels.cpp

```diff
@@ -399,6 +399,149 @@ TEST_F(Csr, AppliesLinearCombinationToDenseMatrix)
 }
 
 
+TEST_F(Csr, AppliesToCsrMatrix)
+{
+    mtx->apply(mtx3_unsorted.get(), mtx2.get());
+    mtx2->sort_by_column_index();
```

Comment:
nit: The sorting is actually part of the Assert in your case (you only use it to know the indices for the other asserts as far as I can tell), so it should be in the group of the ASSERTS (also applies to the other tests in here).